### DISCUSSION TITLE
Get inspired by Picasa

### DISCUSSION THREAD

**Discussion by rodrigopolo** [immich-app/immich#10472] [General]
Picasa was so good that Google acquire it to make it's transition to Google Photos, the very service Immich is trying to emulate:

July 30, 2004 - Google Reveals Details of Picasa Deal
https://www.wsj.com/articles/SB109121493116979168

2016 - We've decided to retire Picasa in order to focus on a single photo service in Google Photos
https://picasa.google.com/

Even though Immich is inspired by Google Photos, I think it should get more inspiration from Picasa and even some from Synology Photos, here are some of the things where Picasa shines:

## Face metadata saved into the files:
![Metadata](https://github.com/immich-app/immich/assets/1511174/f27015aa-8006-40dc-8e3d-1ec9d7e09fb9)

```
[XMP]           RegionAppliedToDimensionsW      : 8192
[XMP]           RegionAppliedToDimensionsH      : 5464
[XMP]           RegionAppliedToDimensionsUnit   : pixel
[XMP]           RegionName                      : Hugo, Paco, Luis, Rodrigo, Carlos
[XMP]           RegionType                      : Face, Face, Face, Face, Face
[XMP]           RegionAreaX                     : 0.444641, 0.594666, 0.376892, 0.286865, 0.510315
[XMP]           RegionAreaY                     : 0.169015, 0.327233, 0.452233, 0.401171, 0.453148
[XMP]           RegionAreaW                     : 0.10437, 0.111938, 0.109985, 0.101318, 0.134399
[XMP]           RegionAreaH                     : 0.188324, 0.201318, 0.198389, 0.18265, 0.241581
[XMP]           RegionAreaUnit                  : normalized, normalized, normalized, normalized, normalized
```

## Sorting
![Sorting](https://github.com/immich-app/immich/assets/1511174/e80ed79e-94cb-4064-91ec-eca8e62892b3)

## Folder or flat view
![folders](https://github.com/immich-app/immich/assets/1511174/aec224e6-243c-4c5b-a024-b41b2547207e)

## Filters by stars, with faces, by type, location and date range
![filters](https://github.com/immich-app/immich/assets/1511174/20e92925-7e44-45fb-ab6e-9b9cc12ede67)

## Folder view and sorting inside each folder
![folder view](https://github.com/immich-app/immich/assets/1511174/5b77e98f-dcda-4cf7-9dc1-27b548bc7aec)

## General features
* Image editor, histogram and general photo info on the left sidebar
* People, location, tags and info on the right sidebar
* Slideshow and comparison tools in the header
* Selection, rating, rotation, tagging, zoom and other actions in the footer
* Face information on roll over, and the possibility for manually adding people
![general features](https://github.com/immich-app/immich/assets/1511174/50852767-c5de-4f0b-98b7-e13009a22d19)

## Global search
Search into folder names, image file names, tags, peoples names, etc:
![Global search](https://github.com/immich-app/immich/assets/1511174/faefefd1-5c99-4e75-9107-88c2e1326e0b)

And there are other features I guess it should consider, some are already under the feature request here, some are not:

* Complete library sharing with optional hidden folders
* Roll over on a face in an image to ee/edit name
* Search by folder name
* Shortcut to jump to date and/or folder by date, or from the info panel
  https://github.com/immich-app/immich/discussions/4115
* Clickable date, camera info, folder for search
* Star and color rating and filtering
  https://github.com/immich-app/immich/discussions/3619
* Folder view
  https://github.com/immich-app/immich/discussions/5409
* Manual face tagging
  https://github.com/immich-app/immich/discussions/7301
* Image rotation, Web Photo Editor
  https://github.com/immich-app/immich/discussions/1695
  https://github.com/immich-app/immich/pull/9575
  https://github.com/immich-app/immich/pull/3271
* General sorting options for views
  https://github.com/immich-app/immich/discussions/10400
  https://github.com/immich-app/immich/discussions/8068
* Organize & search with tags
  https://github.com/immich-app/immich/discussions/8304

Another thing that is missing, is clear guides on how tu upgrade Immich to the latest version, the release notes only show some of the lines that have to be changed. Overall, I love Immich, that is why I strongly believe it should consider get some inspiration in the good old Picasa.

**Comment by rodrigopolo** [comment]
About the jump to a date, here is a workaround using bookmarklets in any web browser, just create a bookmark, edit it, and paste this JS code in the address, it will jump to the date of the current displayed picture based on the HMTL:
```js
javascript:!function()%7Blet%20e%3Ddocument.querySelector(%22%23detail-panel%20%3E%20section%20%3E%20div.px-4.py-4%20%3E%20button%22)%2Ct%3De.innerText.match(%2F(%5Cw%2B%20%5Cd%7B1%2C2%7D%2C%20%5Cd%7B4%7D)%2F)%3Bif(t)%7Blet%20n%3Dnew%20Date(t%5B0%5D)%3Bn.setUTCHours(n.getTimezoneOffset()%2F60)%3Blet%20i%3Dnew%20Date(n.getTime()%2B86399999)%2Co%3DJSON.stringify(%7BtakenAfter%3An.toISOString()%2CtakenBefore%3Ai.toISOString()%7D)%2Cr%3D%60https%3A%2F%2F%24%7Bwindow.location.hostname%7D%2Fsearch%3Fquery%3D%24%7BencodeURIComponent(o)%7D%60%3Bwindow.location.href%3Dr%7Delse%20alert(%22Date%20not%20found%20in%20the%20provided%20string.%22)%7D()%3B
```

This is the JS code:
```js
javascript:(function() {
	const button = document.querySelector("#detail-panel > section > div.px-4.py-4 > button");
	const dateMatch = button.innerText.match(/(\w+ \d{1,2}, \d{4})/);

	if (dateMatch) {
		const takenAfter = new Date(dateMatch[0]);
		takenAfter.setUTCHours(takenAfter.getTimezoneOffset() / 60);

		const takenBefore = new Date(takenAfter.getTime() + 86399999); // 24 hours - 1 millisecond

		const query = JSON.stringify({
			takenAfter: takenAfter.toISOString(),
			takenBefore: takenBefore.toISOString()
		});

		const url = `https://${window.location.hostname}/search?query=${encodeURIComponent(query)}`;
		window.location.href = url;
	} else {
		alert('Date not found in the provided string.');
	}
})();
```

Also, a bookmarklet to go to an specific date:
```js
javascript:!function()%7Bvar%20e%3Dprompt(%22Please%20enter%20a%20date%20(YYYY-MM-DD)%3A%22)%3Blet%20t%3De.match(%2F(%5Cd%7B4%7D%5C-%5Cd%7B02%7D%5C-%5Cd%7B02%7D)%2F)%3Bif(t)%7Blet%20n%3Dnew%20Date(t%5B0%5D)%3Bn.setUTCHours(n.getTimezoneOffset()%2F60)%3Blet%20o%3Dnew%20Date(n.getTime()%2B86399999)%2Ci%3DJSON.stringify(%7BtakenAfter%3An.toISOString()%2CtakenBefore%3Ao.toISOString()%7D)%2Cr%3D%60https%3A%2F%2F%24%7Bwindow.location.hostname%7D%2Fsearch%3Fquery%3D%24%7BencodeURIComponent(i)%7D%60%3Bwindow.location.href%3Dr%7Delse%20alert(%22Date%20not%20found%20in%20the%20provided%20string.%22)%7D()%3B
```

And the JS code:
```js
javascript:(function() {
	var date = prompt('Please enter a date (YYYY-MM-DD):');
	const dateMatch = date.match(/(\d{4}\-\d{02}\-\d{02})/);

	if (dateMatch) {
		const takenAfter = new Date(dateMatch[0]);
		takenAfter.setUTCHours(takenAfter.getTimezoneOffset() / 60);

		const takenBefore = new Date(takenAfter.getTime() + 86399999); // 24 hours - 1 millisecond

		const query = JSON.stringify({
			takenAfter: takenAfter.toISOString(),
			takenBefore: takenBefore.toISOString()
		});

		const url = `https://${window.location.hostname}/search?query=${encodeURIComponent(query)}`;
		window.location.href = url;
	} else {
		alert('Date not found in the provided string.');
	}
})();
```

**Comment by alexhk** [comment]
I too miss Picasa and look forward to Immich getting some of the Picasa simplicity and usability. 

In fact I am still holding on to my old Picasa install on my main library, but only open it every year or so these days.
